apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/DataActivity.java

```diff
@@ -171,20 +178,37 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         mMessageText = findViewById(R.id.data_message);
 
         /* List the app read-only documents. */
-        mAppDocumentListAdapter = new AppDocumentListAdapter(this, new ArrayList<DocumentWrapper<TestDocument>>());
+        mAppDocumentListAdapter = new AppDocumentListAdapter(this, new ArrayList<DocumentWrapper<Map>>());
         mAppDocumentListAdapter.setOnItemClickListener(new AppDocumentListAdapter.OnItemClickListener() {
 
             @Override
             public void onItemClick(int position) {
                 Intent intent = new Intent(DataActivity.this, DocumentDetailActivity.class);
+                DocumentWrapper<Map> document = mAppDocumentListAdapter.getDocument(position);
                 intent.putExtra(DOCUMENT_PARTITION, DefaultPartitions.APP_DOCUMENTS);
-                intent.putExtra(DOCUMENT_ID, mAppDocumentListAdapter.getDocumentByPosition(position));
+                intent.putExtra(DOCUMENT_ID, document.getId());
+                if (document.getError() != null) {
+                    String message = document.getError().getMessage();
+                    if (message.length() > MAX_CONTENT_LENGTH) {
+                        message = message.substring(0, MAX_CONTENT_LENGTH) + "...";
+                        intent.putExtra(DOCUMENT_ERROR, message);
+                        intent.putExtra(DOCUMENT_ERROR_NULL_STATUS,true);
+                    }
+                }
+                else{
+                    intent.putExtra(DOCUMENT_DATE, document.getLastUpdatedDate());
+                    intent.putExtra(DOCUMENT_STATE,document.isFromDeviceCache());
```

Comment:
```suggestion
                    intent.putExtra(DOCUMENT_STATE, document.isFromDeviceCache());
```

Reply:
Actually I noticed several of these missing space changes.  We use the default Android Studio formatter for the most part, which on Mac is cmd-option-L

Just run it and these should be taken care of :)